redis/src/cluster_routing.rs

```diff
@@ -460,39 +458,36 @@ pub enum SlotAddr {
 /// to avoid the need to choose a replica each time
 /// a command is executed
 #[derive(Debug, Eq, PartialEq)]
-pub(crate) struct SlotAddrs([String; 2]);
+pub(crate) struct SlotAddrs {
+    primary: String,
+    replicas: Vec<String>,
+}
 
 impl SlotAddrs {
-    pub(crate) fn new(master_node: String, replica_node: Option<String>) -> Self {
-        let replica = replica_node.unwrap_or_else(|| master_node.clone());
-        Self([master_node, replica])
+    pub(crate) fn new(primary: String, replicas: Vec<String>) -> Self {
+        Self { primary, replicas }
     }
 
     pub(crate) fn slot_addr(&self, slot_addr: SlotAddr) -> &str {
-        if slot_addr == SlotAddr::Replica {
-            self.0[1].as_str()
+        if slot_addr == SlotAddr::Master || self.replicas.is_empty() {
+            self.primary.as_str()
         } else {
-            self.0[0].as_str()
+            self.replicas[0].as_str()
```

Comment:
Why do we always return the first replica?

Reply:
would you prefer to always get a random replica?

Reply:
yes